vstorage/vstorage.go

```diff
@@ -0,0 +1,78 @@
+package vstorage
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+type Vstorage struct {
+	Name string
+}
+
+type Mntent struct {
+	Device string
+	Path   string
+	Type   string
+}
+
+func readMounts(path string) ([]Mntent, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	out := []Mntent{}
+	r := bufio.NewReader(f)
+	for {
+		line, err := r.ReadString('\n')
+		if err == io.EOF {
+			break
+		}
+		fields := strings.Fields(line)
+		if len(fields) < 3 {
+			continue
+		}
+
+		out = append(out, Mntent{
+			Device: fields[0],
+			Path:   fields[1],
+			Type:   fields[2],
+		})
+	}
+	return out, nil
+}
+func (v *Vstorage) Mountpoint() (string, error) {
+	// find out cluster mount point
+	mounts, err := readMounts("/proc/mounts")
+	if err != nil {
+		return "", err
+	}
+	mount := ""
+	for _, m := range mounts {
+		if m.Type == "fuse.vstorage" && m.Device == "vstorage://"+v.Name {
+			mount = m.Path
+			break
+		}
+	}
+	return mount, nil
+}
+
+func (v *Vstorage) Auth(password string) error {
+	auth := exec.Command("/usr/bin/vstorage", "-c", v.Name, "auth-node", "-P")
```

Comment:
1. I believe that if PATH is misconfigured, the absence of vstorage will be a minor problem.
2. If someone can overload PATH, it can overload LD_LIBRARY_PATH and give you another glibc for example, so it doesn't look like a solution for this security problem.

I have seen a lot of old scripts where absolute paths are used. But now this tradition is disappeared.